flasc/data_processing/northing_offset_change_hoger.py

```diff
@@ -0,0 +1,288 @@
+        # through them
+        if is_split_node:
+            stack.append((children_left[node_id], depth + 1))
+            stack.append((children_right[node_id], depth + 1))
+        else:
+            is_leaves[node_id] = True
+
+    # Get the values and the position of the external knots
+    leave_values = values[is_leaves][:, 0, 0]
+    knot_positions = np.sort(threshold[~is_leaves])
+
+    return leave_values, knot_positions
+
+
+def _discretize(x: pd.Series, threshold: int) -> np.ndarray:
+    """Get the class of the knots based on the times they repeat.
+
+    Args:
+        x (pd.Series): Series of knot positions of the trees for the different wind turbines.
+        threshold (int): Threshold used to declare a tree branch.
+
+    Returns:
+        np.ndarray: Classes of the knots.
+    """
+    o = np.argsort(x)  # Get the order of the knots
+    x = x.iloc[o]  # Order the knots in ascending value of their positions
+    y = np.ones(len(x), dtype=np.int32)
+    d = np.diff(x)
+    w = np.where(d >= threshold)[0]
+
+    for i in range(len(d)):
+        y[i + 1] = y[i] + (i in w)
+
+    y = y[np.argsort(o)]
+
+    return y
+
+
+def _shorth_mode(x: pd.Series) -> np.float64:
+    """Estimates the Venter mode through the shorth method for the given data.
+
+    Args:
+        x (pd.Series): Data for which the mode will be estimated
+
+    Returns:
+        np.float64: Mode of the data
+
+    """
+    ny = len(x)
+    k = int(np.ceil(ny / 2) - 1)
+    y = np.sort(x)
+    diffs = y[k:] - y[: (ny - k)]
+    i = np.where(diffs == min(diffs))[0]
+
+    if len(i) > 1:
+        if (np.max(i) - np.min(i)) > (_MODE_LIMIT * ny):
+            warnings.warn(
+                "Encountered a tie, and the difference between minimal and maximal value "
+                f"is > length('x') * {_MODE_LIMIT}.\n The distribution could be multimodal"
+            )
+        i = int(np.mean(i))
+    else:
+        i = i[0]
+
+    mode = np.mean(y[i : (i + k + 1)])
+
+    return mode
+
+
+def _plot_regression(y_data: pd.Series, y_regr: np.ndarray, date_time: pd.Series, ylabel: str):
+    """Function to plot the results of the regression tree.
+
+    Args:
+        y_data (pd.Series): Data used on the regression.
+        y_regr (np.ndarray): Results obtained from the tree regression.
+        date_time (pd.Series): Dates of the original data.
+        ylabel (str): Data that is shown in the plot.
+
+    """
+    fig = plt.figure()
+    sc = plt.scatter(date_time, y_data, c=dates.date2num(date_time), s=5)
+    cbar = fig.colorbar(sc)
+    loc = dates.AutoDateLocator()
+    cbar.ax.yaxis.set_major_locator(loc)
+    cbar.ax.yaxis.set_major_formatter(dates.ConciseDateFormatter(loc))
+    plt.plot(date_time[~y_data.isna()], y_regr, c="tab:red")
+    plt.ylabel(ylabel)
+    plt.grid()
+    plt.tight_layout()
+
+    plt.show()
+
+
+# TODO: Keep these defaults?
+def homogenize(
```

Comment:
Perhaps we'd like a bit more of a descriptive function name (such as `homogenize_wind_directions`, or even `homegenize_HOGER`, or similar)?